woudc_data_registry/models.py

```diff
@@ -325,17 +332,27 @@ class Instrument(base):
 
     def __init__(self, dict_):
         self.station_id = dict_['station_id']
-        self.dataset_name = dict_['dataset_name']
-        self.dataset_level = str(float(dict_['dataset_level']))
         self.dataset_form = str(dict_.get('dataset_form', '1'))  # noqa; defaults to '1' if empty/none/no Key found
-        self.contributor = dict_['contributor']
-        self.project = dict_['project']
 
         self.name = dict_['name']
         self.model = dict_['model']
         self.serial = dict_['serial']
 
-        self.generate_ids()
+        try:
+            self.instrument_id = dict_['instrument_id']
+            self.dataset_id = dict_['dataset_id']
+            self.deployment_id = dict_['deployment_id']
```

Comment:
Should there be a `try/except` clause for these separately? Currently, only `LOGGER.warning()` is given about the `deployment_id` and not about the `instrument_id` and `dataset_id`. 

Or alternatively, update the `LOGGER.warning()` to include about `instrument_id` and `dataset_id` being generated too from other fields.